selenium_tests/lti_emailer/emailer_permission_tests.py

```diff
@@ -0,0 +1,45 @@
+import unittest
+from ddt import ddt, data, unpack
+
+from selenium_tests.base_test_case import get_xl_data
+from selenium_tests.lti_emailer.emailer_base_test_case import CANVAS_ADD_USERS
+from selenium_tests.lti_emailer.page_objects.canvas_masquerade_page_object import \
+    CanvasMasqueradePageObject
+from selenium_tests.lti_emailer.emailer_base_test_case import \
+    EmailerBaseTestCase
+
+
+@ddt
+class EmailerPermissionTests(EmailerBaseTestCase):
+    @data(*get_xl_data(CANVAS_ADD_USERS))
+    @unpack
+    def test_roles_access(self, user_id, given_access):
+        # This test masquerades as users in roles specified defined in
+        # CANVAS_ADD_USERS (Excel spreadsheet) then validates if the users
+        # are granted/denied access based on their role.
+
+        emailer_main_page = self.emailer_main_page
+        masquerade_page = CanvasMasqueradePageObject(self.driver, self.CANVAS_BASE_URL)
+        masquerade_page.masquerade_as(user_id)
+        masquerade_page.confirm_masquerade()
```

Comment:
Ah yes, I forgot this part! Good catch. That said, the `confirm_masquerade()` should be _part of_ the `masquerade_as()` function -- in other words, this line should move to the `masquerade_as()` function, at the end, and not be needed in test scripts. (Test scripts shouldn't need to know how masquerade works in Canvas.)